mantidproject/mantid
# Fix sorting in `CalculatePolynomialBackground`

**Description of work.**
Fixes sorting logic deep in `CalculatePolynomialBackground` that was showcasing different behaviour on different platforms, possibly due to equality check of doubles.



**To test:**
Run the new system test which should fail on current master on macOS.




*There is no associated issue, since it's a small change.*




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
